### PR TITLE
Feature Integration Specs

### PR DESCRIPTION
**Pull request by AbuTalha3**
**In this PR I have**
- Made sure that the N+1 problem is solved when fetching all posts and their comments for a user.
- Do not use the bullet gem in this exercise - instead, analyze the logs as shown in the given article
- In my PR description include the screenshots of your app's logs before and after the fix.

**Here are the SS**

image

image

### CODE REVIEW

File: spec/features/users_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      expect(page).to have_css("img[src*='https://unsplash.com/photos/2.jpg']")
+      expect(page).to have_css("img[src*='https://unsplash.com/photos/3.jpg']")
+    end
+
+    it 'should redirect to that users show page when clicking on the username' do
+      visit users_path
+      click_link @user1.name
+      expect(page).to have_current_path(user_path(@user1))
+    end
+  end
```
Comment (danmainah) [inline spec/features/users_index_spec.rb]: - [ ] @Team Great work adding tests for the user index page, however its missing the test highlighted below listed below
image

File: spec/features/users_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    it 'should have the button to display all posts' do
+      visit user_path(@user1)
+      expect(page).to have_content('View All Posts')
+    end
+    it "should redirect me to the user's post's index page when clicking on the 'View All Posts' button" do
+      visit user_path(@user1)
+      click_link 'View All Posts'
+      expect(page).to have_current_path(user_posts_path(@user1))
+    end
+  end
```
Comment (danmainah) [inline spec/features/users_show_spec.rb]: - [ ] @Team Great work adding tests for the user index page, however, its missing the test highlighted below listed below
image

File: spec/features/posts_index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+      visit user_posts_path(@user1)
+      expect(page).to have_content(@post1.likes_counter)
+    end
+
+    it "should redirect me to that post's show page when clicking on the post's title" do
+      visit user_posts_path(@user1)
+      click_link @post1.title
+      expect(page).to have_current_path(user_post_path(@user1, @post1))
+    end
+  end
```
Comment (danmainah) [inline spec/features/posts_index_spec.rb]: - [ ] @Team Great work adding tests for the user index page, however, its missing the test highlighted below listed below
image

- [ ] @Team Kindly check on some of your tests they are failing as shown below:
image

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @AbuTalha3 and @Megagig , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️  Fixed the N + 1 problem.
✔️  Added tests for the users and posts index nad show pages.


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by brainconnect93 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @AbuTalha3 & @Megagig, 👋 👋  

Your project is complete! Great job for following the project requirement🥇and implementing the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 




## Good Points 👍

- Descriptive Pull Request title & summary. ✔️
- integration specs for views and fix n+1 problems ✔️
- Good use of Git flow ✔️

## Optional changes

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brainconnect93 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**